### ISSUE TITLE
[MNT]: Refactor kwargs of Tick constrctor

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#26008]
### Summary

Currently, the Tick constructor smashes together kwargs specific to it, kwargs for the tickmarkers, for the labels, and for the gridline:
```python
    def __init__(
        self, axes, loc, *,  # global
        size=None,  # tickmarker property
        width=None,  # tickmarker property
        color=None,  # global
        tickdir=None,  # tickmarker (not a property)
        pad=None,  # global
        labelsize=None,  # label property
        labelcolor=None,  # label property
        labelfontfamily=None,  # label property
        zorder=None,  # global
        gridOn=None,  # gridline (property: ~visible)
        tick1On=True,  # tick (property, but specific to each tick)
        tick2On=True,  # tick (property, but specific to each tick)
        label1On=True,  # label (property, but specific to each label)
        label2On=False,  # label (property, but specific to each label)
        major=True,  # global
        labelrotation=0,  # label property
        grid_color=None,  # grid property
        grid_linestyle=None,  # grid property
        grid_linewidth=None,  # grid property
        grid_alpha=None,  # grid property
        **kwargs,  # grid properties
    ):
```
The remaining kwargs are implicitly assumed to all start with "grid_" and further interpreted as grid properties (via `grid_kw = {k[5:]: v for k, v in kwargs.items()}`) -- note that this prefix is internally added by `Axis.grid()` to kwargs passed to it.

### Proposed fix

We should at least check whether remaining kwargs indeed start with "grid_" and error out if that's not the case (Probably few people actually manually construct Ticks, but Tick(abcdecolor="k")) should certainly not be interpreted as Tick(grid_color="k") anyways.)

It may make sense to try to group the kwargs a bit more and not try to repeat all relevant properties in the Tick constructor, e.g. (not writing out the defaults)
```python
def __init__(
    self, axes, loc, *,
    major, tickdir, pad, color, zorder,  # global
    tick_props, tick1On, tick2On,  # tick properties
    label_props, label1On, label2On,  # label properties
    grid_props,  # grid properties
):
```
(probably kwargs could be normalized in this form as soon as one populates Axis._major_tick_kw/_minor_tick_kw)

**Comment by madeira-dev** [comment]
Hi @anntzer, I read this issue and thought about a possible way to fix the first part of it, related to check if remaining kwargs are starting with "grid_".

I thought about something like this:
![Screenshot from 2023-05-30 19-44-34](https://github.com/matplotlib/matplotlib/assets/69992340/fcd597be-8ce2-4073-ad4e-74dcecb01ab9)

Where right after defining grid_kw I would check for any other kwargs.

Please let me know if this is an appropriate approach.

**Comment by anntzer** [comment]
#26037 is a step in the right direction, although the "extra grouping" still needs to be considered.

**Comment by joooeey** [comment]
The [docs](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.grid.html) say `xdata` and `ydata` are allowed keyword arguments. Those can be useful, e.g. to [extend a grid beyond the axes limits](https://stackoverflow.com/a/76383402/4691830) or create diagonal gridlines.

A possible fix for this particular issue is outlined in the [commit linked here](https://github.com/matplotlib/matplotlib/compare/main...joooeey:matplotlib:main).

**Comment by joooeey** [comment]
The refactoring proposed here would also open the door for another enhancement that would allow passing more label props with `set_tick_params` such as the backgroundcolor which AFAIK can only be set on the individual labels right now.

**Comment by has2k1** [comment]
`Axis.set_tick_params` does too much and it keeps a record it uses to replay the previous settings along with any new settings. But because it is a high level method, its record gets stale when properties are changed on the artists (ticklines, labels & gridlines) directly.

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
ax.scatter([1, 2, 3], [1, 2, 3])

# Make xaxis ticklines red and thick
for t in ax.xaxis.get_major_ticks():
    t.tick1line.set(markeredgecolor="red", markeredgewidth=5)

# Make the axis ticklines longer
ax.xaxis.set_tick_params(size=8)  # Undoes the markedgewidth!
```

![set_tick_params](https://github.com/matplotlib/matplotlib/assets/780341/ab10d287-6173-4be3-b10d-f8737f18da6d)

**Comment by has2k1** [comment]
> As for the specific case, changing markeredgewidth seems unnecessary. I have not looked into the details. It may be possible to fix that, but this way of working is fragile by design. 

Yes it is fragile, but I have a theming system where I want access to more properties on the artists than `set_tick_params` allows. But I also use it because some properties e.g. _tick direction_ are not conveniently composed to be set directly.

**Comment by timhoffm** [comment]
For your case, I suggest to do as much as possible via `tick_params` and only afterwards change individual ticks properties that cannot be configured otherwise. For example, your code above can do everything in `tick_params`:
```
ax.xaxis.set_tick_params(color="red", width=5, size=8)
```

Also, what are the parameters, you are missing in `tick_params`? We are fundamentally open to extend it.

**Comment by joooeey** [comment]
I think `tick_params` should simply support all parameters accepted by labels, ticks, and gridlines. This is necessary because there is no alternative to achieve the same functionality for interactive plots. For parameters that can be passed to both labels and ticks, it should be possible to pass them individually (and perhaps also together as with the `color` kwarg).

If this was a new method, I'd want an API with three separate methods or one where you can pass three dictionaries. But one has to work with the mess that is there. Perhaps phasing out `set_tick_params()` and replacing with `format_ticklines()`, `format_ticklabels()`, and `format_grid()` is an option.
